providers/redis/redis.go

```diff
@@ -109,7 +121,12 @@ func (p *RedisProvider) DeleteCacheParameterGroup(ctx context.Context, instanceI
 	return err
 }
 
-func (p *RedisProvider) Update(ctx context.Context, instanceID string, params providers.UpdateParameters) error {
+func (p *RedisProvider) UpdateReplicationGroup(ctx context.Context, instanceID string, params providers.UpdateReplicationGroupParameters) error {
+	replicationGroupID := GenerateReplicationGroupName(instanceID)
+	return p.modifyReplicationGroup(ctx, replicationGroupID, params.PreferredMaintenanceWindow)
+}
+
+func (p *RedisProvider) UpdateParams(ctx context.Context, instanceID string, params providers.UpdateParamGroupParameters) error {
```

Comment:
Urrrrrrrgh I don't get the point of this "abstraction" when `UpdateParamGroupParameters` just has a single field of `map[string]string`, and now `UpdateReplicationGroup` isn't symmetrical with `UpdateParams` because the former has knowledge of its arguments and the latter has no knowledge of its possible arguments (which raises the question of what the purpose of this abstraction is). But asking you to fix it would either be telling you to write your bit the bad way too or asking you to rewrite other peoples code.

Also trying to figure out if "cache parameter group" (as in `modifyCacheParameterGroup`) "param group parameters" (as in `UpdateParamGroupParameters`) and just "params" (as in `UpdateParams`) are the same thing and if so why they need several different names.

Reply:
Is this a general issue that should go on the backlog, or a blocker for this feature?

Reply:
Could you just choose one of the existing names for the cache.. parameter.. group.. params.. ? Instead of just "Params", because this further compounds the confusion.

Reply:
Believe this is now resolved?